test/functional/tests/cache_ops/test_multistream_seq_cutoff.py

```diff
@@ -0,0 +1,250 @@
+#
+# Copyright(c) 2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+import os
+import random
+from time import sleep
+import pytest
+from api.cas import casadm
+from api.cas.cache_config import CacheMode, SeqCutOffPolicy, CacheModeTrait
+from core.test_run_utils import TestRun
+from storage_devices.disk import DiskTypeSet, DiskTypeLowerThan, DiskType
+from test_tools.dd import Dd
+from test_tools.disk_utils import Filesystem
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.os_utils import Udev
+from test_utils.size import Size, Unit
+
+random_thresholds = random.sample(range(1028, 1024 ** 2, 4), 3)
+random_stream_numbers = random.sample(range(2, 256), 3)
+
+
+@pytest.mark.parametrizex("streams_number", [5, 256] + random_stream_numbers)
+@pytest.mark.parametrizex("threshold",
+                          [Size(1, Unit.MebiByte), Size(1, Unit.GibiByte)]
+                          + [Size(x, Unit.KibiByte) for x in random_thresholds])
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_multistream_seq_cutoff_functional(threshold, streams_number):
+    """
+    title: Functional test for multistream sequential cutoff
+    description: |
+        Testing if amount of data written to cache and core is correct after running sequential
+        writes from multiple streams with different sequential cut-off thresholds.
+    pass_criteria:
+        - Amount of blocks written to cache is equal to amount set with sequential cutoff threshold
+        - Amount of blocks written in pass-through is equal to io size run after gaining the
+        sequential cutoff threshold
+    """
+    with TestRun.step("Start cache and add core device."):
+        cache_disk = TestRun.disks['cache']
+        core_disk = TestRun.disks['core']
+
+        cache = casadm.start_cache(cache_disk, CacheMode.WB, force=True)
+        Udev.disable()
+        core = cache.add_core(core_disk)
+
+    with TestRun.step(f"Set seq-cutoff policy to always, threshold to {threshold} "
+                      f"and reset statistics counters."):
+        core.set_seq_cutoff_policy(SeqCutOffPolicy.always)
+        core.set_seq_cutoff_threshold(threshold)
+        core.reset_counters()
+
+    with TestRun.step(f"Run {streams_number} I/O streams with number of sequential writes equal to "
+                      f"seq-cutoff threshold value minus one 4k block."):
+        kib_between_streams = 100
+        offsets = [o for o in range(0, int(streams_number * (threshold.get_value(Unit.KibiByte)
+                                                             + kib_between_streams)),
+                                    int(threshold.get_value(Unit.KibiByte)) + kib_between_streams)]
```

Comment:
low readability
Can You rearrange this? `0` is unnecessary and may buy some space.

Reply:
actually 0 is necessary, because without it the third parameter will be used as a "stop", not as a "step" param for range() method and the range method does not take keyword arguments.

Reply:
Sorry, You're right.